### ISSUE TITLE
Error arising from unexpected integer-typed datum_ids

### ISSUE THREAD

**Issue by danielballan** [bluesky/bluesky-browser#28]
From SST via @mrakitin 

```
Traceback (most recent call last):
  File "c:\users\greateyes\event-model\event_model\__init__.py", line 501, in event
    datum_doc = self._datum_cache[datum_id]
KeyError: 1

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\users\greateyes\intake-bluesky\intake_bluesky\core.py", line 463, in read_partition
    self.filler('event', event)
  File "c:\users\greateyes\event-model\event_model\__init__.py", line 599, in __call__
    return super().__call__(name, doc, validate)
  File "c:\users\greateyes\event-model\event_model\__init__.py", line 71, in __call__
    output_doc = getattr(self, name)(doc)
  File "c:\users\greateyes\event-model\event_model\__init__.py", line 507, in event
    raise err_with_key from err
event_model.UnresolvableForeignKeyError: Event with uid fb5bfc8e-597d-4e23-8620-1650a3fb348a refers to unknown Datum datum_id 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\greateyes\bluesky-browser\bluesky_browser\viewer\viewer.py", line 263, in run
    for name, doc in self.entry().read_canonical():
  File "c:\users\greateyes\intake-bluesky\intake_bluesky\core.py", line 434, in read_canonical
    for name, doc in self.read_partition(i):
  File "c:\users\greateyes\intake-bluesky\intake_bluesky\core.py", line 467, in read_partition
    if '/' in datum_id:
TypeError: argument of type 'int' is not iterable
```

**Comment by danielballan** [comment]
Run Start UID 0372d961

**Comment by mrakitin** [comment]
We made some progress with https://github.com/bluesky/intake-bluesky/pull/56 fix. Here is what I observed today:
<img width="703" alt="image" src="https://user-images.githubusercontent.com/13209176/59642562-86a79500-9133-11e9-9d5d-0cec729ed1c4.png">
I.e. one of the images was filled, but another - wasn't. Looks like one loop is missing somewhere.

**Comment by mrakitin** [comment]
That's where it failed: https://github.com/bluesky/event-model/blob/dcfdb49c15b33e6a0832d71c879841e81686172f/event_model/__init__.py#L504; the screenshot above displays the value of the `doc` variable just before the failure.

**Comment by mrakitin** [comment]
The error is:
```py
Traceback (most recent call last):
  File "C:\Users\greateyes\event-model\event_model\__init__.py", line 498, in event
    datum_doc = self._datum_cache[datum_id]
KeyError: '14ac7e0e-b55e-4eb4-a9b3-25a59ba53921/0'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\greateyes\intake-bluesky\intake_bluesky\core.py", line 570, in read_partition
    self.filler('event', event)
  File "C:\Users\greateyes\event-model\event_model\__init__.py", line 596, in __call__
    return super().__call__(name, doc, validate)
  File "C:\Users\greateyes\event-model\event_model\__init__.py", line 71, in __call__
    output_doc = getattr(self, name)(doc)
  File "C:\Users\greateyes\event-model\event_model\__init__.py", line 504, in event
    raise err_with_key from err
event_model.UnresolvableForeignKeyError: Event with uid 7fd15f7e-32e5-418c-858b-2ce09a0d8fbb refers to unknown Datum datum_id 14ac7e0e-b55e-4eb4-a9b3-25a59ba53921/0

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\greateyes\event-model\event_model\__init__.py", line 498, in event
    datum_doc = self._datum_cache[datum_id]
KeyError: '7a9eb4a0-f8ed-4013-b308-50d5124581fb/0'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm Community Edition 2018.3.4\helpers\pydev\_pydev_bundle\pydev_monkey_qt.py", line 180, in _new_run
    return self._original_run()
  File "C:\Users\greateyes\bluesky-browser\bluesky_browser\viewer\viewer.py", line 259, in run
    for name, doc in self.entry().read_canonical():
  File "C:\Users\greateyes\intake-bluesky\intake_bluesky\core.py", line 539, in read_canonical
    for name, doc in self.read_partition(i):
  File "C:\Users\greateyes\intake-bluesky\intake_bluesky\core.py", line 586, in read_partition
    self.filler('event', event)
  File "C:\Users\greateyes\event-model\event_model\__init__.py", line 596, in __call__
    return super().__call__(name, doc, validate)
  File "C:\Users\greateyes\event-model\event_model\__init__.py", line 71, in __call__
    output_doc = getattr(self, name)(doc)
  File "C:\Users\greateyes\event-model\event_model\__init__.py", line 504, in event
    raise err_with_key from err
event_model.UnresolvableForeignKeyError: Event with uid 7fd15f7e-32e5-418c-858b-2ce09a0d8fbb refers to unknown Datum datum_id 7a9eb4a0-f8ed-4013-b308-50d5124581fb/0
```

**Comment by danielballan** [comment]
Suggested fix is pushed as https://github.com/bluesky/intake-bluesky/pull/56/commits/43a521edf1633c5c750999212c8b7ecccce72aad. Needs to be tested at SST.

**Comment by mrakitin** [comment]
This is a great fix! I got a better result now:
![image](https://user-images.githubusercontent.com/13209176/59724577-62fa5280-91f8-11e9-9a5d-a7d63dfc5ded.png)

This is what I see with `intake-bluesky`
```python
In [20]: next(hdr_intake)
Out[20]:
('event',
 {'descriptor': '9a2dda7e-b7c7-4b2d-9e4e-fd68319a0b97',
  'uid': '7fd15f7e-32e5-418c-858b-2ce09a0d8fbb',
  'time': 1559146958.8257315,
  'seq_num': 1,
  'data': {'Synced_saxs_image': array([[    8929,   157074,   176136, ...,    92298,    98567,   104586],
          [    9565,   157563,   195632, ...,   113746,   124458,   124965],
          [    4427,   158560,   211486, ...,   131186,   140908,   141301],
          ...,
          [16771863,    98400,   201927, ...,   217701,   216994,    76366],
          [     327,   100870,   203414, ...,   215682,   212422,    45258],
          [    4644,    99648,   193807, ...,   209051,   205952,    39495]],
         dtype=uint32),
   'Synced_saxs_stats1_total': 14020288401.0,
   'Synced_waxs_image': array([[    4194,    47972,   134809, ...,   152955,    67956,     8257],
          [    4233,    47647,   135181, ...,   169622,    68096,     8317],
          [    4312,    47597,   134923, ...,   182853,    67823,     8381],
          ...,
          [   15940,    15929,    15933, ..., 16776650,      251,     3835],
          [   15803,    15650,    16233, ..., 16776423,      244,     3917],
          [   14971,    15917,    16348, ..., 16776595,      379,     3984]],
         dtype=uint32),
   'Synced_waxs_stats1_total': 165354802663.0,
   'Beamline Energy_energy': 708.0848685,
   'Beamline Energy_energy_setpoint': 708.0848685,
   'Beamline Energy_monoen_setpoint': 320.0,
   'Beamline Energy_monoen_readback': 708.0848685,
   'Beamline Energy_monoen_done': 4,
   'Beamline Energy_monoen_stop_signal': 1.0,
   'Beamline Energy_epugap': 100004.40000000001,
   'Beamline Energy_epugap_user_setpoint': 100000.0,
   'Izero Mesh Drain Current': -7.3318e-13},
  'timestamps': {'Synced_saxs_image': 1559146955.8502033,
   'Synced_saxs_stats1_total': 1559147011.925474,
   'Synced_waxs_image': 1559146955.842224,
   'Synced_waxs_stats1_total': 1559147011.533999,
   'Beamline Energy_energy': 1559066830.9447062,
   'Beamline Energy_energy_setpoint': 1559066830.9457343,
   'Beamline Energy_monoen_setpoint': 1557159538.822714,
   'Beamline Energy_monoen_readback': 1559132724.368967,
   'Beamline Energy_monoen_done': 1557854822.504653,
   'Beamline Energy_monoen_stop_signal': 1553093091.039981,
   'Beamline Energy_epugap': 1559120510.373802,
   'Beamline Energy_epugap_user_setpoint': 1558557608.365888,
   'Izero Mesh Drain Current': 1559147009.710622},
  'filled': {'Synced_waxs_image': '7a9eb4a0-f8ed-4013-b308-50d5124581fb/0',
   'Synced_saxs_image': '14ac7e0e-b55e-4eb4-a9b3-25a59ba53921/0'}})
```

That's how to reproduce it:
```py
In [22]: %history
cat = mn.BlueskyMongoCatalog(db.mds._db, db.fs._db, handler_registry=db.reg.handler_reg)
import intake_bluesky.mongo_normalized as mn
cat = mn.BlueskyMongoCatalog(db.mds._db, db.fs._db, handler_registry=db.reg.handler_reg)
cat
entry = cat['0372d961']
hdr_intake = entry.read_canonical()
hdr_intake
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
next(hdr_intake)
```

**Comment by danielballan** [comment]
Hm. So it looks like the Artist is receiving data (the limits on the colorbar seem to be adapting as expected) but is maybe not being drawn.

**Comment by danielballan** [comment]
The last bug illustrated above is an unrelated problem, fixed by #36. I think we can close this now.
